src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -488,12 +490,18 @@ class TaskExecutor {
       boost::optional<ModelMetrics> cur_model_metric) {
     std::shared_ptr<ModelContainer> processing_container =
         active_containers_->get_model_replica(completed_msg.model_,
-                                              completed_msg.container_id_);
+                                              completed_msg.replica_id_);
```

Comment:
This was the critical change. Active containers is a map of maps that keys containers by <VersionedModelID, replica_id> but we were looking them up by <VersionedModelID, container_id>. As soon as replica_id and container_id diverged, the container wouldn't be found and a nullptr was returned. We also failed to check if `processing_container` was null, so we tried to access a null ptr and caused a segfault.